solutions/fully-configurable/main.tf

```diff
@@ -0,0 +1,212 @@
+}
+
+module "buckets" {
+  count   = local.create_buckets ? 1 : 0
+  source  = "terraform-ibm-modules/cos/ibm//modules/buckets"
+  version = "8.19.3"
+  bucket_configs = [
+    {
+      bucket_name                         = local.logs_bucket_name
+      kms_key_crn                         = local.kms_encryption_enabled ? local.kms_key_crn : null
+      kms_guid                            = local.kms_encryption_enabled ? module.existing_kms_crn_parser[0].service_instance : null
+      kms_encryption_enabled              = local.kms_encryption_enabled
+      region_location                     = var.region
+      resource_instance_id                = var.existing_cos_instance_crn
+      management_endpoint_type_for_bucket = var.management_endpoint_type_for_bucket
+    },
+    {
+      bucket_name                         = local.metrics_bucket_name
+      kms_key_crn                         = local.kms_encryption_enabled ? local.kms_key_crn : null
+      kms_guid                            = local.kms_encryption_enabled ? module.existing_kms_crn_parser[0].service_instance : null
+      kms_encryption_enabled              = local.kms_encryption_enabled
+      region_location                     = var.region
+      resource_instance_id                = var.existing_cos_instance_crn
+      management_endpoint_type_for_bucket = var.management_endpoint_type_for_bucket
+      skip_iam_authorization_policy       = true
+    }
+  ]
+}
+
+data "ibm_cos_bucket" "existing_logs_bucket" {
+  count                = local.create_buckets ? 0 : 1
+  bucket_name          = var.existing_logs_cos_bucket_name
+  resource_instance_id = var.existing_cos_instance_crn
+  bucket_region        = var.existing_logs_cos_bucket_region
+  bucket_type          = var.existing_logs_cos_bucket_type
+}
+
+data "ibm_cos_bucket" "existing_metrics_bucket" {
+  count                = local.create_buckets ? 0 : 1
+  bucket_name          = var.existing_metrics_cos_bucket_name
+  resource_instance_id = var.existing_cos_instance_crn
+  bucket_region        = var.existing_metrics_cos_bucket_region
+  bucket_type          = var.existing_metrics_cos_bucket_type
+}
+
+module "existing_kms_crn_parser" {
+  count   = var.existing_kms_instance_crn != null ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.existing_kms_instance_crn
+}
+
+module "existing_kms_key_crn_parser" {
+  count   = local.use_kms_module ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = local.kms_key_crn
+}
+
+module "kms" {
+  count                       = local.use_kms_module ? 1 : 0
+  source                      = "terraform-ibm-modules/kms-all-inclusive/ibm"
+  version                     = "4.20.0"
+  create_key_protect_instance = false
+  region                      = var.region
```

Comment:
this region need to be parsed from the KMS instance CRN. It might be a different region to the Cloud logs instance